pkg/cache/clusterqueue.go

```diff
@@ -306,6 +328,16 @@ func (c *clusterQueue) inactiveReason() (string, string) {
 	return kueue.ClusterQueueActiveReasonReady, "Can admit new workloads"
 }
 
+func (c *clusterQueue) isTASViolated() bool {
+	if !features.Enabled(features.TopologyAwareScheduling) || len(c.tasFlavors) == 0 {
+		return false
+	}
+	return c.HasParent() ||
+		c.Preemption.WithinClusterQueue != kueue.PreemptionPolicyNever ||
+		len(c.multiKueueAdmissionChecks) > 0 ||
+		len(c.provisioningAdmissionChecks) > 0
```

Comment:
Does this mean that we support TAS collaboration with other third-party AdmissionCheck controllers?

Reply:
It means we don't support validation OOTB, if you use an external admission check with TAS you need to validate yourself. We could maybe make the mechanism under https://github.com/kubernetes-sigs/kueue/issues/3106 flexible enough to cover TAS.

Reply:
> We could maybe make the mechanism under https://github.com/kubernetes-sigs/kueue/issues/3106 flexible enough to cover TAS.

Let's investigate the possibility of this in the future.
But, I guess that we can reject all CQs with any admissionCheck names (`.spec.admissionChecks` or `.spec.admissionChecksStrategy`), right?

Reply:
If we did that, then it would be stronger condition - meaning that TAS cannot be used with some external ACs, even if there is no issue. For example, the AC checks team's budget.

Reply:
> For example, the AC checks team's budget.

That makes sense. In that case, we should reject the CQs with AC only related to the scheduling like ProvisioningRequest and MultiKueue. Thanks.